sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient.cs

```diff
@@ -524,6 +524,144 @@ public virtual RecognizeBusinessCardsOperation StartRecognizeBusinessCardsFromUr
+        /// </summary>
+        /// <param name="invoice">The stream containing the one or more invoices to recognize values from.</param>
+        /// <param name="recognizeInvoicesOptions">A set of options available for configuring the recognize request.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A <see cref="RecognizeInvoicesOperation"/> to wait on this long-running operation.  Its <see cref="RecognizeInvoicesOperation.Value"/> upon successful
+        /// completion will contain the extracted invoices.</returns>
+        public virtual RecognizeInvoicesOperation StartRecognizeInvoices(Stream invoice, RecognizeInvoicesOptions recognizeInvoicesOptions = default, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNull(invoice, nameof(invoice));
+
+            recognizeInvoicesOptions ??= new RecognizeInvoicesOptions();
+
+            using DiagnosticScope scope = Diagnostics.CreateScope($"{nameof(FormRecognizerClient)}.{nameof(StartRecognizeInvoices)}");
+            scope.Start();
+
+            try
+            {
+                FormContentType contentType = recognizeInvoicesOptions.ContentType ?? DetectContentType(invoice, nameof(invoice));
+
+                Response response = ServiceClient.AnalyzeInvoiceAsync(contentType, invoice, recognizeInvoicesOptions.IncludeFieldElements, recognizeInvoicesOptions.Locale, cancellationToken);
+                string location = ClientCommon.GetResponseHeader(response.Headers, Constants.OperationLocationHeader);
+
+                return new RecognizeInvoicesOperation(ServiceClient, Diagnostics, location);
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Recognizes values from one or more invoices.
+        /// <para>See <a href="https://aka.ms/formrecognizer/businesscardfields"/> for a list of available fields on an invoice.</para>
+        /// </summary>
+        /// <param name="invoicesUri">The absolute URI of the remote file to recognize values from.</param>
+        /// <param name="recognizeInvoicesOptions">A set of options available for configuring the recognize request.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A <see cref="RecognizeInvoicesOperation"/> to wait on this long-running operation.  Its <see cref="RecognizeInvoicesOperation.Value"/> upon successful
+        /// completion will contain the extracted invoices.</returns>
+        public virtual async Task<RecognizeInvoicesOperation> StartRecognizeInvoicesFromUriAsync(Uri invoicesUri, RecognizeInvoicesOptions recognizeInvoicesOptions = default, CancellationToken cancellationToken = default)
```

Comment:
in the stream method, we called the invoice parameter `invoice` and here it is plural `invoicesUri`. Should it be `invoiceUri`? 

I noticed in Python we tend to use plurality in the method name and then use the singular with input parameters. Not sure if this is an established pattern or just what we do in Form Recognizer.

Reply:
Good catch! I will change to singular as all other endpoints are managing it that way.

hmm yeah we do the same in .NET ... I do not know but I won't be able to change the parameter names for the 3.0 release though, so I think we could stay with consistency across library

Reply:
Yeah same, I'm siding with consistency too

Reply:
:(